### PR TITLE
pay contracts: new contracts to support axelar bridge

### PR DESCRIPTION
**Pull request by andrewliu08**

### CODE REVIEW

File: packages/contract/script/Constants.s.sol [hunk 1]
```diff
@@ -325,6 +345,19 @@ function _getAxlUsdcAddress(uint256 chainId) pure returns (address) {
     return address(0);
 }
 
+function _getAxlUSDCAddress(uint256 chainId) pure returns (address) {
```
Comment (nalinbhardwaj) [inline packages/contract/script/Constants.s.sol:348]: this is a dupe function? see just over this

File: packages/contract/src/pay/DaimoPayCCTPBridger.sol [hunk 2]
```diff
@@ -18,58 +18,86 @@ import "../../vendor/cctp/ICCTPTokenMessenger.sol";
...
-    mapping(uint256 chainId => uint32 domain) public cctpDomainMapping;
+    // Map destination chainId to CCTP domain and the bridge token address on the
+    // destination chain.
+    mapping(uint256 toChainId => CCTPBridgeRoute bridgeRoute)
+        public bridgeRouteMapping;
 
-    event DomainAdded(uint256 indexed chainId, uint32 domain);
+    event BridgeRouteAdded(
+        uint256 indexed toChainId,
+        CCTPBridgeRoute bridgeRoute
```
Comment (andrewliu08) [inline packages/contract/src/pay/DaimoPayCCTPBridger.sol:39]: is it fine to have a struct in the event like this?

File: packages/contract/script/pay/DeployDaimoPayAxelarBridger.s.sol [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        ) {
+            chainIds = new uint256[](1);
+            bridgeRoutes = new DaimoPayAxelarBridger.AxelarBridgeRoute[](1);
+
+            chainIds[0] = BSC_MAINNET;
+
+            for (uint32 i = 0; i < chainIds.length; ++i) {
+                uint256 fee = block.chainid == POLYGON_MAINNET
+                    ? 3000000000000000000 // 3.0 POLYGON
+                    : 300000000000000; // 0.0003 ETH
```
Comment (nalinbhardwaj) [inline packages/contract/script/pay/DeployDaimoPayAxelarBridger.s.sol:82]: maybe 0.0005 ETH? base to binance charged me like $0.8

File: packages/contract/script/pay/DeployDaimoPayAxelarBridger.s.sol [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+            chainIds[5] = POLYGON_MAINNET;
+
+            for (uint32 i = 0; i < chainIds.length; ++i) {
+                bridgeRoutes[i] = DaimoPayAxelarBridger.AxelarBridgeRoute({
+                    destChainName: _getAxelarChainName(chainIds[i]),
+                    bridgeTokenIn: _getAxlUsdcAddress(block.chainid),
+                    bridgeTokenOut: _getAxlUsdcAddress(chainIds[i]),
+                    bridgeTokenOutSymbol: "axlUSDC",
+                    receiverContract: axelarReceiver,
+                    fee: 2_000_000_000_000_000 // 2 * 10^15 = 0.002 BNB
```
Comment (nalinbhardwaj) [inline packages/contract/script/pay/DeployDaimoPayAxelarBridger.s.sol:111]: maybe factor out into a function `getAxelarFeeByChainID(chainID)` that gets ~$1 in chain's native token?

File: packages/contract/src/pay/DaimoPayAxelarBridger.sol [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+            fromToken: inToken,
+            fromAmount: inAmount,
+            toChainId: toChainId,
+            toAddress: toAddress,
+            toToken: outToken,
+            toAmount: outAmount
+        });
+    }
+
+    receive() external payable {}
```
Comment (nalinbhardwaj) [inline packages/contract/src/pay/DaimoPayAxelarBridger.sol:318]: can make `sendToChain` payable?

File: packages/daimo-common/src/daimoPay.ts [hunk 6]
```diff
@@ -54,6 +54,15 @@ export interface DaimoPayOrderItem {
   image?: string;
 }
 
+export const zBridgeTokenOutOption = z.object({
+  token: zAddress.transform((a) => getAddress(a)),
+  amount: z.string().transform((a) => BigInt(a)),
```
Comment (nalinbhardwaj) [inline packages/daimo-common/src/daimoPay.ts:59]: `zBigintStr`?

File: packages/daimo-common/src/daimoPay.ts [hunk 7]
```diff
@@ -124,8 +133,10 @@ export type DaimoPayDehydratedOrder = {
 export type DaimoPayHydratedOrder = {
   mode: DaimoPayOrderMode.HYDRATED;
   id: bigint;
-  handoffAddr: Address;
-  destMintTokenAmount: DaimoPayTokenAmount;
+  intentAddr: Address;
+  bridgeTokenOutOptions: DaimoPayTokenAmount[];
+  bridgeTokenOutAddr: Address | null;
```
Comment (nalinbhardwaj) [inline packages/daimo-common/src/daimoPay.ts:138]: `selectedBridgeTokenOut`?

**Pull request merged**